riva/proto/riva_nmt.proto

```diff
@@ -21,12 +24,88 @@ service RivaTranslation {
 
   // Translate text to text, from a source to a target language.  Currently source and target language fields is required, along with the model name.
   // Multiple texts may be passed per request up to the given batch size for the model, which is set at translation pipeline creation time.
-rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
+  rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
 
   // Lists the available language pairs and models names to be used for TranslateText
   rpc ListSupportedLanguagePairs(AvailableLanguageRequest) returns (AvailableLanguageResponse) {}
+
+  //streaming speech to text translation api.
+  rpc StreamingTranslateSpeechToText(stream StreamingTranslateSpeechToTextRequest)
+        returns (stream StreamingTranslateSpeechToTextResponse) {}
+
+  //offline speech to text translation api.
+  //rpc TranslateSpeechToText(TranslateSpeechToTextRequest) returns (TranslateSpeechToTextResponse) {}
+
+  //Returns the parameters for the configurated pipelines.
+  //rpc GetTranslateSpeechToTextConfig(GetTranslateSpeechToTextConfigRequest) returns (GetTranslateSpeechToTextConfigResponse) {}
+
+  rpc StreamingTranslateSpeechToSpeech(stream StreamingTranslateSpeechToSpeechRequest)
+    returns (stream StreamingTranslateSpeechToSpeechResponse) {}
+
+  //offline speech to text translation api.
+  //rpc TranslateSpeechToSpeech(TranslateSpeechToSpeechRequest) returns (TranslateSpeechToSpeechResponse) {}
+
+
 }
 
+/*
+* Configuration for Translate S2S.  reuse existing protos from other services.
+*/
+message StreamingTranslateSpeechToSpeechConfig {
+   nvidia.riva.asr.StreamingRecognitionConfig asrConfig = 1;  //from riva_asr.proto
+   SynthesizeSpeechConfig ttsConfig = 2; //from riva_tts.proto
+   string source_language_code = 3;
+   string target_language_code = 4;
+   AudioEncoding t = 5;
+}
+
+/*
+* Streaming translate speech to speech used to configure the entire pipline for speech translation.  This can be
+* be backed by a cascade of ASR, NMT, TTS models or an end to end model
+*
+*/
+message StreamingTranslateSpeechToSpeechRequest {
+   oneof streaming_request {
+       StreamingTranslateSpeechToSpeechConfig config = 1;
+       bytes audio_content = 2;
+   }
+}
+
+message SynthesizeSpeechConfig {
```

Comment:
This should probably be moved to riva_tts.proto.  And TTS refactored to accommodate the config.

Reply:
TTS needs a revisit, out of scope for this change.  TTS operates per request and doesnt have a streaming concept,  This config defines how we call multiple TTS requests within a S2S stream.